### DISCUSSION TITLE
如何对孪生网络特征进行可视化

### DISCUSSION THREAD

**Discussion by IceHowe** [frgfm/torch-cam#206] [Q&A]
如何对孪生网络特征进行可视化，输入是两张图像，backbone经过了两次，每张图像经过一次，例如：
```pyhon
self.backbone = resnet18()

forward(x1,x2):
    fx1 = self.backbone(x1)
    fx2 = self.backbone(x2)
```

**Comment by frgfm** [comment]
Hello @IceHowe :wave:

I'm sorry, but I had to use Google Translate to understand your request so my understanding might be limited :/ 

Now regarding your question about visualizing a twin network:
- I'm not sure how, in your example, this is a twin network
- with your snippet, I suggest using the CAM extractor twice: after the first forward, and after the last

I'm pretty sure I didn't get your actual request, so feel free to specify :sweat_smile: 
Would you mind translating to English for your next message please? (or if you prefer French, German, Spanish)
Cheers :v:

**Comment by vegeckn** [comment]
I have the same problem with you. And the method throws out an error: forward() missing 1 required positional argument: 'x2'. I do not know how to solve this problem. Do you solve this ?
